kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/codec/KafkaRequestDecoder.java

```diff
@@ -73,15 +75,15 @@ protected Frame decodeHeaderAndBody(ChannelHandlerContext ctx, ByteBuf in, final
         LOGGER.debug("{}: {} downstream correlation id: {}", ctx, apiKey, correlationId);
         RequestHeaderData header = null;
         final ByteBufAccessorImpl accessor;
-        inbodundMessageCounter.increment();
+        inboundMessageCounter.increment();
         var decodeRequest = decodePredicate.shouldDecodeRequest(apiKey, apiVersion);
         LOGGER.debug("Decode {}/v{} request? {}, Predicate {} ", apiKey, apiVersion, decodeRequest, decodePredicate);
         boolean decodeResponse = decodePredicate.shouldDecodeResponse(apiKey, apiVersion);
         LOGGER.debug("Decode {}/v{} response? {}, Predicate {}", apiKey, apiVersion, decodeResponse, decodePredicate);
         short headerVersion = apiKey.requestHeaderVersion(apiVersion);
         if (decodeRequest) {
             decodedMessagesCounter.increment();
-            Metrics.payloadSizeBytesUpstreamSummary(apiKey, apiVersion).record(length);
+            Metrics.payloadSizeBytesUpstreamSummary(apiKey, apiVersion, clusterName).record(length);
```

Comment:
its almost like we should have a `metrics` instance that know the name of its virtual cluster rather than a static method... 

Thats not a refactoring for this PR however.